src/include/uapi/ipc/topology.h

```diff
@@ -289,16 +267,4 @@ struct sof_ipc_pipe_comp_connect {
 	uint32_t sink_id;
 } __attribute__((packed));
 
-/* create new component kpb - SOF_IPC_TPLG_KPB_NEW */
-struct sof_ipc_comp_kpb {
```

Comment:
~~In general it's OK for me. I only think `sof_ipc_process_type ` shall be removed as we don't use it at all.~~

After a talk with validation team I think we should keep components structures here in `src/include/uapi/ipc/topology.h` and remove them from components header files. Validation team is referenced as code-owner of this file and any change in common structures is automaticly visible by them so they can quickly provide a feedback and/or apply changes in the slim driver.
So in case of KPB for example we should move `struct sof_kpb_config` here. Same thing for detector, selector etc.